packages/geo/src/lib/datasource/shared/datasources/wms-datasource.ts

```diff
@@ -81,17 +81,43 @@ export class WMSDataSource extends DataSource {
     if (!options.sourceFields || options.sourceFields.length === 0) {
       options.sourceFields = [];
     }
+    const ogcFilterableDataSourceOptions: OgcFilterableDataSourceOptions = this.options;
+    if (sourceParams.layers.split(',').length > 1 && this.options
+    && ogcFilterableDataSourceOptions.ogcFilters
```

Comment:
You can make a variable out of ogcFilterableDataSourceOptions.ogcFilters. This will shorten your ifs a lot.